com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -238,7 +239,7 @@ define([
                     $scope.edit_data[role_id]['funders'] = [];
 
                     // Split data from the stored funder contact IDs
-                    var funder_contact_ids = data['funder'].split("|");
+                    var funder_contact_ids = $filter('isNotUndefined')(data['funder'].split("|"));
```

Comment:
If you want to separate it from the controller, please move those methods into a service. When (if)  we need to use those in a view, we can introduce $filters that can reuse the service methods.